byway/src/main/java/com/google/sps/servlets/PlacesServlet.java

```diff
@@ -52,4 +74,13 @@ private String formatLocation(PlaceType location) {
     place = place.replace('_', ' ');
     return place;
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String tripKeyString = request.getParameter("tripId");
+    ArrayList<String> interests = gson.fromJson(request.getReader(), ARRAYLIST_STRING);
+    Trip trip = Trip.getTrip(datastore, tripKeyString);
```

Comment:
Might be good to check that the trip is not null, and set a `NOT_FOUND` status if it is missing.

Reply:
Updated